temporalio/contrib/openai_agents/_temporal_model_stub.py

```diff
@@ -0,0 +1,160 @@
+from __future__ import annotations
+
+from temporalio import workflow
+
+with workflow.unsafe.imports_passed_through():
+    from typing import Any, AsyncIterator, Sequence, Union, cast
+
+    from agents import (
+        AgentOutputSchema,
+        AgentOutputSchemaBase,
+        ComputerTool,
+        FileSearchTool,
+        FunctionTool,
+        Handoff,
+        Model,
+        ModelResponse,
+        ModelSettings,
+        ModelTracing,
+        Tool,
+        TResponseInputItem,
+        WebSearchTool,
+    )
+    from agents.items import TResponseStreamEvent
+    from openai.types.responses.response_prompt_param import ResponsePromptParam
+
+    from temporalio.contrib.openai_agents.invoke_model_activity import (
+        ActivityModelInput,
+        AgentOutputSchemaInput,
+        FunctionToolInput,
+        HandoffInput,
+        ModelActivity,
+        ModelTracingInput,
+        ToolInput,
+    )
+
+
+class _TemporalModelStub(Model):
+    """A stub that allows invoking models as Temporal activities."""
+
+    def __init__(self, model_name: Union[str, None], **kwargs) -> None:
+        self.model_name = model_name
+        self.kwargs = kwargs
+
+    async def get_response(
+        self,
+        system_instructions: Union[str, None],
+        input: Union[str, list][TResponseInputItem],  # type: ignore
+        model_settings: ModelSettings,
+        tools: list[Tool],
+        output_schema: Union[AgentOutputSchemaBase, None],
+        handoffs: list[Handoff],
+        tracing: ModelTracing,
+        *,
+        previous_response_id: Union[str, None],
+        prompt: ResponsePromptParam | None,
```

Comment:
This syntax isn't 3.9-compatible; need to use `Optional` (should be a failing test)